### ISSUE TITLE
Macro style meta-architecture instantiation

### ISSUE THREAD

**Issue by vertexclique** [bastion-rs/bastion#3]
Instead of writing a bunch of boilerplate code. People can able to do this:

```rust
supervisor! { OneForAll,
    child! {
        |context, msg: Box<dyn Message>| {
            // some thing over here
        }
    }

    children! { 100,
        |context, msg: Box<dyn Message>| {
            // some thing over here too
        }
    }
}
```

**Comment by Stupremee** [comment]
I will implement these macros

**Comment by vertexclique** [comment]
Fixed by #140
